daemon/cleanup.go

```diff
@@ -30,9 +41,21 @@ func registerSigHandler() <-chan struct{} {
 		for s := range sig {
 			log.WithField("signal", s).Info("Exiting due to signal")
 			pidfile.Clean()
+			Clean()
+			if option.Config.FlannelUninstallOnExit {
+				for _, ep := range endpointmanager.GetEndpoints() {
+					ep.DeleteBPFProgramLocked()
```

Comment:
This is being called without the endpoint mutex being acquired - if this is intended, document why.